corehq/apps/hqcase/views.py

```diff
@@ -154,7 +154,12 @@ def _handle_bulk_fetch(request):
 
 def _handle_list_view(request):
     try:
-        res = get_list(request.domain, request.GET.dict())
+        params = request.GET.dict()
+        if 'username' in params and 'api_key' in params:
```

Comment:
Does it matter that `username` and `api_key` only get removed if they're both present, rather than removing them individually?

Reply:
I was a little torn on that, open to second opinions. If both are provided, then I can be somewhat confident that the user used them for auth.  If only one is present, then they didn't.  This way if a client uses a more normal auth method and try to filter cases by username, then they'll get the normal error message `"username" is not a valid parameter`, rather than silently ignoring the param.  The downside is that this introduces a slight logical difference between authentication methods, which I also don't like.